go-controller/pkg/ovn/pods.go

```diff
@@ -326,8 +343,11 @@ func (oc *Controller) addLogicalPort(pod *kapi.Pod) error {
 	}
 
 	if annotation == nil {
-		gwIfAddr := util.GetNodeGatewayIfAddr(nodeSubnet)
-		routes, gwIP, err := getRoutesGatewayIP(pod, gwIfAddr)
+		hybridOverlayExternalGw, err := oc.waitForHybridOverlayExternalGwAnnotation(pod.Namespace)
```

Comment:
FTR I don't like making us hold up the pod creation for namespace creation, but it is necessary for this patch. @dave-tucker I suspect you will need to come up with some type of rework on this for your workqueue/modularity patches

Reply:
I wonder if @girishmg you think this should be conditional based on if hybrid overlay is enabled?

Reply:
@trozet thanks for the note. Yes, make it conditional based on whether hybrid overlay is enabled or not will be good.

Reply:
sounds good, i will amend in a bit